unit-tests/log/test-multi-thread-logs.cpp

```diff
@@ -0,0 +1,231 @@
+{
+    std::string line;
+    for (int i = 0; i < (number_of_iterations * number_of_threads) - 1; ++i)
+    {
+        getline(in, line);
+    }
+    return line;
+}
+
+// inputs are use as 2 points that define a line:
+// first_time_ms, first_value : x1, y1
+// second_time_ms, second_value : x2, y2
+// value_in_log : y3
+// return value will be x3
+int get_log_ideal_time(int first_time_ms, int first_value, int second_time_ms, int second_value, int value_in_log)
+{
+    int ideal_time = value_in_log * (second_time_ms - first_time_ms) / (second_value - first_value) +
+        (first_time_ms * (second_value / (second_value - first_value))) - second_time_ms * (first_value / (second_value - first_value));
+    
+    return ideal_time;
+}
+
+std::pair<unsigned int, int> get_next_50_lines_time_val(std::ifstream& ifs)
+{
+    unsigned int avg_time_ms = -1;
+    int avg_value = -1;
+    for (int i = 0; i < 50; ++i) {
+        std::string line;
+        getline(ifs, line);
+        avg_time_ms += get_time_from_line(line);
+        avg_value += stoi(line.substr(line.find_last_of(" ")));
+    }
+    avg_time_ms /= 50;
+    avg_value /= 50;
+    return std::make_pair(avg_time_ms, avg_value);
+}
+
+//----------------------- HELPER METHODS END ---------------------------
+
+TEST_CASE("async logger", "[log][async_log]")
+{
+    size_t n_callbacks = 0;
+    auto callback = [&](rs2_log_severity severity, rs2::log_message const& msg)
+    {
+        ++n_callbacks;
+        std::cout << msg.raw();
+    };
+    
+    auto func = [](int required_value) {
+        int iterations = 0;
+        auto start = std::chrono::steady_clock::now();
+        
+        while (iterations < number_of_iterations)
+        {
+            std::stringstream ss;
+            ss << "atomic integer = " << ++atomic_integer;
+            auto start_ms = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch());
+            rs2::log(RS2_LOG_SEVERITY_DEBUG, ss.str().c_str());
+            std::chrono::milliseconds ms = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch());
+            std::chrono::milliseconds delta_ms = ms - start_ms;
+            if (delta_ms > max_time) max_time = delta_ms;
+            if (delta_ms < min_time) min_time = delta_ms;
+            avg_time += delta_ms;
+            start_ms = ms;
+            ++iterations;
+        }
+    };
+    const char* log_file_path = ".//multi-thread-logs.log";
+    try {
+        remove(log_file_path);
+    }
+    catch (...) {}
+    rs2::log_to_file(RS2_LOG_SEVERITY_DEBUG, log_file_path);
+
+    std::vector<std::thread> threads;
+
+    for (int i = 0; i < number_of_threads; ++i)
+    {
+        threads.push_back(std::thread(func, i + 1));
+    }
+    
+    for (auto&&t : threads)
+    {
+        if (t.joinable()) t.join();
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(5000));
```

Comment:
What for?

Reply:
This is to give time to the logs to be dispatched by the logger thread.

Reply:
Not sure I understand... 5 seconds? What logger thread?

Reply:
logger thread: Since the logger is in asynchronous mode, there is a thread created by the logger. This thread pulls the logs from a queue and dispatches them (to file, console...).
The 5 seconds must be too long, it might be shortened.